Corona-Warn-App/src/main/java/de/rki/coronawarnapp/datadonation/safetynet/AttestationContainer.kt

```diff
@@ -0,0 +1,56 @@
+package de.rki.coronawarnapp.datadonation.safetynet
+
+import de.rki.coronawarnapp.appconfig.SafetyNetRequirements
+import de.rki.coronawarnapp.datadonation.safetynet.SafetyNetException.Type
+import de.rki.coronawarnapp.server.protocols.internal.ppdd.PpacAndroid
+import okio.ByteString.Companion.toByteString
+
+internal data class AttestationContainer(
+    private val ourSalt: ByteArray,
+    private val report: SafetyNetClientWrapper.Report
+) : DeviceAttestation.Result {
+    override val accessControlProtoBuf: PpacAndroid.PPACAndroid
+        get() = PpacAndroid.PPACAndroid.newBuilder().apply {
+            salt = ourSalt.toByteString().base64()
+            safetyNetJws = report.jwsResult
+        }.build()
+
+    override fun requirePass(reqs: SafetyNetRequirements) {
+        if (reqs.requireBasicIntegrity && !report.basicIntegrity) {
+            throw SafetyNetException(Type.BASIC_INTEGRITY_REQUIRED, "Requirement 'basicIntegrity' not met.")
```

Comment:
I read in the docs, that the attestation response optionally returns:
> `advice`: A suggestion for how to get a device back into a good state.

What do you think about adding the `advice` property to the `Report` and add it to the exception messages here, if available?

Reply:
That's a nice thought. Will do.